### ISSUE TITLE
Conduits

### ISSUE THREAD

**Issue by LemADEC** [LemADEC/WarpDrive#40]
Conduits are your all around transportation block adapted to the mod. They're a pre-requitei for fly-by ships with weapon integration:
- [] Energy conduit transfer energy from the mod's reactor(s) to other systems. It's notably more efficient than using EU/RF to transfer energy
- [] Inventory conduit transfer items for the shipyard
- [] Plasma conduit transfer fluid/gases. Typically weapon energy from producer to emitter

Conduits work as a network and do instant transfer. There's no pump or pressure, nor progressive transmission.
By extension, in a later version, WarpPlasma could be transfered from reactor to warp injector into the warp coils to enable warping (as per StarTrek lore).

**Comment by lellasone** [comment]
This seems like it could also be achieved using forge fluids and pre-existing liquid transfer systems which would give players more options in terms of how they manage there plasma/weapon energy/ect.

**Comment by LemADEC** [comment]
Yes, energy/fluid transport are covered by an handful of mods out there. We've however 2 issues with those systems:
1- it won't work in fly-by mode, or it would be a pain to integrate (as they use direct block access, we would need heavy reflection or a new dimension just to simulate the ships)
2- for capital ships and Starbase, most of those implementation are fairly CPU intensive for client (Mekanism & BC are FPS killer) or server (TE is pretty heavy on TPS), or won't scale well (ExtraUtilities notably).
3- I've yet to see an EnderIO-like with ComputerCraft or OpenComputer cables integrated.

The idea remains to keep our blocks compatible with other transportation mods but encourage them to use ours to build a flyable ship or improve transfer efficiency. For example, the Warp core energy transfer would be much more efficient to transfer with the Warp conduit.

**Comment by TheNumenorean** [comment]
To done extent also you want to force player to have warp cores on the ships themselves. If you could pump warp plasma through ender io having ship based energy production becomes practically worthless

**Comment by LemADEC** [comment]
Actually, it becomes a valid solution for a 'battery' powered ship, we just need to require enough warp energy to prevent abuse (unlike MFFS which consumes almost nothing to sustain the field, for example).

**Comment by TheNumenorean** [comment]
I personally just don't like the idea of wireless 100% efficient power. That doesn't mean batteries, that means infinite power source, because a player can have a really large warp core at their home base, with none of the space constraints or necessary protections of having a ship based one. It also means detecting ships is hard or impossible.

**Comment by TheNumenorean** [comment]
Should a ship be able to warp off of batteries? It kiddos defeats the purpose I feel

**Comment by LemADEC** [comment]
I like the idea of the one shot emergency warp.
To use the warp, even without the core, that ship still needs coils of sort.
In my mind, warp energy is supposed to leave a signature behind you, same way as a rocket propulsion leaves heat & smoke behind.
See the starmap ticket once I remember to post it.

**Comment by TheNumenorean** [comment]
I'm ok with that, but it should be only possible to use local storage, and it should only work a few times. Consider if someone built their ships with AE. They could store an enormous amount of liquid for infinite time.
